test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/SelectExpandFunctionalTests.cs

```diff
@@ -1414,6 +1414,71 @@ public void DollarItinFilterInsideSelectShouldReferenceResourcePathEntity()
+                (
+                    clause.SelectedItems.First() as PathSelectItem // $select=RelatedSSNs(...)
+                ).OrderByOption;
+
+            IEdmTypeReference typeReference = (orderByClause.Expression as NonResourceRangeVariableReferenceNode).TypeReference;
+
+            // Assert
+            orderByClause.Expression.ShouldBeNonResourceRangeVariableReferenceNode(ExpressionConstants.This);
+            Assert.Equal(OrderByDirection.Ascending, orderByClause.Direction); // If you don't specify direction, it's Ascending by default.
+            Assert.Equal("Edm.String", typeReference.Definition.FullTypeName()); // RelatedSSNs is a collection of strings.
+        }
+
+        [Fact]
+        public void DollarThisinFilterInsideSelectShouldReferenceSelectedItem()
+        {
+            // Arrange & Act
+            // People?$select=RelatedSSNs($filter=endswith($this,'xyz'))
+            SelectExpandClause clause = RunParseSelectExpand("RelatedSSNs($filter=endswith($this,'xyz'))", "", HardCodedTestModel.GetPersonType(), HardCodedTestModel.GetPeopleSet());
+
+            IEdmTypeReference typeReference =
+                (
+                    (
+                        (
+                            (
+                                clause.SelectedItems.First() as PathSelectItem // $select=RelatedSSNs(...)
+                            ).FilterOption.Expression as SingleValueFunctionCallNode // $filter=endswith($this,'xyz')
+                        ).Parameters.First() as ConvertNode // $this
+                    ).Source as NonResourceRangeVariableReferenceNode
+                ).TypeReference;
+
+            // Assert
+            Assert.Equal("Edm.String", typeReference.Definition.FullTypeName()); // RelatedSSNs is a collection of strings.
+        }
+
+        [Fact]
+        public void DollarThisinFilterInsideSelectInsideExpandShouldReferenceSelectedItem()
+        {
+            // Arrange & Act
+            // People?$expand=MyDog($select=Nicknames($filter=startswith($this, 'blu')))
+            SelectExpandClause clause = RunParseSelectExpand("", "MyDog($select=Nicknames($filter=startswith($this, 'blu')))", HardCodedTestModel.GetPersonType(), HardCodedTestModel.GetPeopleSet());
+            IEdmTypeReference typeReference =
+                (
+                    (
+                        (
+                            (
+                                (
+                                    clause.SelectedItems.First() as ExpandedNavigationSelectItem // $expand=MyDog(...)
+                                ).SelectAndExpand.SelectedItems.First() as PathSelectItem // $select=Nicknames(...)
+                            ).FilterOption.Expression as SingleValueFunctionCallNode // $filter=startswith($this,'blu')
+                        ).Parameters.First() as ConvertNode // $this
+                    ).Source as NonResourceRangeVariableReferenceNode
+                ).TypeReference;

```

Comment:
It seems not readable. 

Consider to refactor 

var selectItem = Assert.Single(clause.SelectedItems);
ExpandedNavigationSelectItem expanded = Assert.IsType<ExpandedNavigationSelectItem>(selectItem);
...

Reply:
Done.
#Resolved